src/components/Wallet/WalletConnectProvider.tsx

```diff
@@ -81,14 +83,27 @@ const extensionProviders: WalletExtensionProvider[] = [
 ]
 
 export const WalletConnectProvider: FC<Props> = ({ children }) => {
+  const [initialState, setInitialState] = useState<State | undefined>(undefined)
+
+  useEffect(() => {
+    if (typeof window !== 'undefined') {
+      const storedState = localStorage.getItem('wagmi.state')
+      if (storedState) {
+        setInitialState(JSON.parse(storedState))
+      }
+    }
+  }, [])
```

Comment:
Remove nesting of if statements where possible
```suggestion
  useEffect(() => {
    if (typeof window === 'undefined') return
    const storedState = localStorage.getItem('wagmi.state')
    if (!storedState) return
    setInitialState(JSON.parse(storedState))
  }, [])
```